solar-system3.rb

```diff
@@ -0,0 +1,162 @@
+  end
+
+  def educate
+    puts "#{@name} is a planet that is #{@size} twinkies in diameter. It #{@rings} rings, so sometimes it feels insecure. It is #{distance_from_the_sun} leagues from the sun and takes #{year_length} readings of \"War and Peace\" to complete its orbit."
+  end
+
+end
+
+##################################################################################################
+
+def dist_btw_planets(selected_planet, other_planets) #a is the planet of interest, b is an array of all the planets
+  other_planets.each do |planet|
+    distance = (selected_planet.distance_from_the_sun - planet.distance_from_the_sun).abs
+    unless selected_planet == planet
+    puts "#{selected_planet.name} is #{distance} leagues from #{planet.name}."
+    end
+  end
+end
+
+##################################################################################################
+
+class SolarSystem
+  attr_accessor :planets
+  attr_reader :age
+
+  def initialize(planets, age) #planets arg must be an array of Planets
+    @planets = planets
+    @age = age
+  end
+
+  def add(planet)
+    @planets << planet
+  end
+
+  def local_year(planet) #planet arg must be an instance of Planet
+    @age / planet.year_length
+  end
+
+  def list_planets
+    list = ""
+    (1..(@planets.length)).each do |i|
+      list += "#{i}. #{@planets[i - 1].name}\n"
+    end
+    return list
+  end
+end
+
+##################################################################################################
+
+# Used during make_planet to force user to input an integer when assigning quantitative Planet attributes
+def force_int(string)
+  until string == string.to_i.to_s
+    print "Please enter an integer: "
+    string = gets.chomp
+  end
+  return string.to_i
+end
+
+def make_planet
+  new_planets = []
+  puts "Want to make a new planet? y/n"
+  input = gets.chomp.downcase
+  until input == "n"
+    print "What is your planet's name?: "
+    name = gets.chomp.downcase
+    print "What is the diameter of your planet in miles?: "
+    size = force_int(gets.chomp)
+    print "Does your planet have rings? y/n: "
+    rings = gets.chomp.downcase
+    until rings == "y" || rings == "n"
+      print "Please answer y or n: "
+      rings = gets.chomp.downcase
+    end
+    rings == "y" ? rings = "has" : rings = "does not have"
+    print "In miles, how far is you planet from the sun?: "
+    distance_from_the_sun = force_int(gets.chomp)
+    print "In years, how long does it take your planet to complete its orbit?: "
+    year_length = force_int(gets.chomp)
+
+    new_planet = Planet.new(name, size, rings, distance_from_the_sun, year_length)
+    new_planets << new_planet
+    puts "Want to make a new planet? y/n"
+    input = gets.chomp.downcase
+  end
+  return new_planets
+end
+
+##################################################################################################
+
+# Call make_planet and store the array it returns
+
+new_planets = make_planet
```

Comment:
The program's flow _may_ be better if you don't call `make_planet` here and instead initialize it with an empty array